src/statusbar.ts

```diff
@@ -40,7 +41,9 @@ class Statusbar {
 
     this.config = Config.get ();
 
-    if ( !this.config.oauthToken ) return vscode.window.showErrorMessage ( 'You need to provide an OAuth token via the "githubNotificationsBell.oauthToken" setting' );
+    if ( !this.oauthToken ) {
```

Comment:
This line here uses the value from `initBell`. Is there a reason why you're getting config here again and not just setting `this.config = Config.get ()` in the `initBell ()`?
I assume that accounts for the possibility for user to update the config and then this method picks up the config updates correctly.

Should I ensure `this.oauthToken` is "re-fetched" from the config (or env) in here?

Reply:
> I assume that accounts for the possibility for user to update the config and then this method picks up the config updates correctly.

I think that's the case, I don't remember clearly.

Just running `this.config.oauthToken || process.env. GITHUB_NOTIFICATIONS_TOKEN` every time instead of caching this value should be enough.